packages/bitcore-node/src/providers/chain-state.ts

```diff
@@ -190,6 +193,49 @@ export class InternalStateProvider implements CSP.IChainStateService {
     Storage.apiStreamingFind(WalletAddressModel, query, { limit }, stream);
   }
 
+  async streamMissingWalletAddresses(params: CSP.StreamWalletMissingAddressesParams) {
+    const { chain, network, pubKey, stream } = params;
+    const wallet = await WalletModel.collection.findOne({ pubKey });
+    const walletId = wallet!._id;
+    const query = { chain, network, wallets: walletId, spentHeight: { $gte: 0 } };
+    const cursor = CoinModel.collection.find(query);
+    const seen = {};
+    const stringifyWallets = (wallets: Array<ObjectId>) => wallets.map(w => w.toHexString());
+    const allMissingAddresses = new Array<string>();
+    let totalMissingValue = 0;
+    const missingStream = cursor.pipe(
+      through2(
```

Comment:
@jguthrie7 this is what I was talking about yesterday, new Transform() can do this?